src/s3path.jl

```diff
@@ -210,45 +228,73 @@ end
 function FilePathsBase.walkpath(fp::S3Path; kwargs...)
     # Select objects with that prefix
     objects = s3_list_objects(fp.config, fp.bucket, fp.key; delimiter="")
+    root = joinpath(fp, "/")
 
     # Construct a new Channel using a recursive internal `_walkpath!` function
-    return Channel(ctype=typeof(fp)) do chnl
-        _walkpath!(fp, fp, Iterators.Stateful(objects), chnl; kwargs...)
+    return Channel(ctype=typeof(fp), csize=128) do chnl
```

Comment:
Again, sizing the channel had a modest improvement when scanning thousands of S3 objects.

```
julia> @benchmark collect(walkpath($fp)) seconds=60
BenchmarkTools.Trial: 13 samples with 1 evaluation.
 Range (min … max):  4.573 s …    5.472 s  ┊ GC (min … max): 0.22% … 0.19%
 Time  (median):     4.817 s               ┊ GC (median):    0.16%
 Time  (mean ± σ):   4.857 s ± 240.414 ms  ┊ GC (mean ± σ):  0.11% ± 0.10%

  ▁   ▁ ▁▁▁▁     ▁    █    ▁    █                          ▁
  █▁▁▁█▁████▁▁▁▁▁█▁▁▁▁█▁▁▁▁█▁▁▁▁█▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁█ ▁
  4.57 s         Histogram: frequency by time         5.47 s <

 Memory estimate: 25.51 MiB, allocs estimate: 469434.
 ```

size=128:
```
julia> @benchmark collect(walkpath($fp)) seconds=60
BenchmarkTools.Trial: 14 samples with 1 evaluation.
 Range (min … max):  4.197 s …    4.756 s  ┊ GC (min … max): 0.22% … 0.17%
 Time  (median):     4.429 s               ┊ GC (median):    0.19%
 Time  (mean ± σ):   4.461 s ± 171.700 ms  ┊ GC (mean ± σ):  0.11% ± 0.10%

  ▁     ▁     █   ▁  ▁▁      ▁    ▁▁             █▁        ▁
  █▁▁▁▁▁█▁▁▁▁▁█▁▁▁█▁▁██▁▁▁▁▁▁█▁▁▁▁██▁▁▁▁▁▁▁▁▁▁▁▁▁██▁▁▁▁▁▁▁▁█ ▁
  4.2 s          Histogram: frequency by time         4.76 s <

 Memory estimate: 25.37 MiB, allocs estimate: 465806.
```